### PR TITLE
Adds tests for searching feature

### LINKED ISSUE #5237: Add spec for new tag searching functionality

### PR DESCRIPTION
**Pull request by gfiorav**
Fixes #5237

CR please @Kartones :)

### CODE REVIEW

**Comment by jesusvazquez** [comment]
retest this please

File: spec/models/visualization/tags_spec.rb [hunk 1]
```diff
@@ -49,7 +49,37 @@
...
+      manolo_escobar_tag = 'Manolo Escobar'
+
+
+      v1 = Visualization::Member.new(random_attributes(user_id: @user_mock.id, name: 'v1', locked:true, tags: [manolo_tag, manoloescobar_tag, manolo_escobar_tag])).store
+      v2 = Visualization::Member.new(random_attributes(user_id: @user_mock.id, name: 'v2', locked:false, tags: [manolo_tag])).store
+      v3 = Visualization::Member.new(random_attributes(user_id: @user_mock.id, name: 'v3', locked:false, tags: [manolo_escobar_tag, manoloescobar_tag])).store
 
+      vqb1 = Carto::VisualizationQueryBuilder.new.with_tags([manolo_tag.upcase])
+      
+      vqb1.build.map(&:id).should include v1.id
```
Comment (Kartones) [inline spec/models/visualization/tags_spec.rb]: why doing 3 times the search? just storing the first `vqb1.build.map(&:id)` inside a variable allows to reuse without rebuilding it once and again

**Comment by Kartones** [comment]
:+1: except for the N extra searches, please remove them and good to go!

File: spec/models/visualization/tags_spec.rb (hunk 1 above)
Comment (gfiorav) [inline spec/models/visualization/tags_spec.rb]: Nice! Will do

**Pull request merged**